src/js/tools/vendor/metro/utils.ts

```diff
@@ -75,8 +80,20 @@ export const createDefaultMetroSerializer = (): MetroSerializer => {
       return code;
     }
 
+    let sourceMapStringFunction;
+    if (typeof sourceMapString === 'function') {
+      sourceMapStringFunction = sourceMapString;
+    } else if (typeof (sourceMapString as NewSourceMapStringExport).sourceMapString === 'function') {
```

Comment:
can `(sourceMapString as NewSourceMapStringExport)` be undefined here in a future version of metro/RN?